### PR TITLE
Refactor styled components for Responsive design with Example Form

### PR DESCRIPTION
**Pull request by cadezacharias-hype**
Closes https://github.com/CrocSwap/ambient-ts-app/issues/3083

Creates an example form to be used as a reference in creating new forms. 

Refactored the common styled components quite a bit to be more modular. Should do a quick UI check of the app to ensure everything is the same. 

Start and nav to:
http://localhost:3000/template/form
https://deploy-preview-3080--ambient-finance.netlify.app/template/form

image

Responsive:

image

### CODE REVIEW

File: src/pages/InitPool/FormExample.tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+                            <IoIosCheckmarkCircle />
+                        </FlexContainer>
+                    </FlexContainer>
+                    {/* Body */}
+                    <FlexContainer
+                        gap={16}
+                        justifyContent='space-around'
+                        flexDirection='column'
+                        maxWidth='500px'
+                        lg={{
```
Comment (cadezacharias-hype) [inline src/pages/InitPool/FormExample.tsx:269]: @Jadessu @emilydrakesmith @benwolski notice this responsive styling with the new components.  It functions like tailwind where the base styles apply to all breakpoints and then the tags `sm, md, lg, xl, xxl` all apply for that size and up.  So here, we say that it's vertical with `column` flex and then from `lg` and up it becomes horizontal with `row` flex.

**Review by Jadessu — APPROVED** [review]
Just finished a thorough review and looks good to me! 
Whether we decide to host the template where it is currently is another discussion.

**Comment by benwolski** [comment]
@cadezacharias-hype, as we discussed in the project meeting, the example form route should only be accessible on local builds.

**Review by surudhb — APPROVED** [review]
looks great!

couple nits: we might not need some of those eslint-disable lines anymore

**Pull request merged**